src/shmem_internal.h

```diff
@@ -54,12 +54,74 @@ extern int shmem_internal_thread_level;
         }                                                               \
     } while (0)
 
+#define SHMEM_ERR_CHECK_ARG_POSITIVE(arg)                               \
+    do {                                                                \
+        if ((arg) <= 0) {                                               \
+            fprintf(stderr, "ERROR: %s(): Argument %s must be positive (%ld)\n", \
+                    __func__, #arg, (long) arg);                        \
+            shmem_runtime_abort(100, "OpenSHMEM exited in error");      \
+        }                                                               \
+    } while (0)
+
+#define SHMEM_ERR_CHECK_ACTIVE_SET(PE_start, logPE_stride, PE_size)                                     \
+    do {                                                                                                \
+        int shmem_err_check_active_stride = 1 << logPE_stride;                                          \
+        if (PE_start < 0 || logPE_stride < 0 || PE_size < 0 ||                                          \
+            PE_start + (PE_size - 1) * shmem_err_check_active_stride > shmem_internal_num_pes) {        \
+            fprintf(stderr, "ERROR: %s(): Invalid active set (PE_start = %d, logPE_stride = %d, PE_size = %d)\n", \
+                    __func__, PE_start, logPE_stride, PE_size);                                         \
+            shmem_runtime_abort(100, "OpenSHMEM exited in error");                                      \
+        }                                                                                               \
+        if (! (shmem_internal_my_pe >= PE_start &&                                                      \
```

Comment:
I prefer it as-is, since the first check validates that the active set args are valid and the second validates that the calling PE is a member of the active set.